### ISSUE TITLE
Duplicate services after update to 0.8.4-rc1

### ISSUE THREAD

**Issue by shantanugadgil** [hashicorp/nomad#4376]
### Nomad version
0.8.4-rc1 (upgraded from 0.8.3)

### Operating system and Environment details
CentOS 7.5 + updates

### Issue
My test cluster consists of Consul 1.1.0 and Nomad 0.8.4-rc1 (previously 0.8.3).
I have configured Consul and Nomad as systemd services
A few **system** jobs and a lot of **service** jobs.

Just like usual, I ran my "update_all_clients" script.

My "update" script basically is a loop which:
* sshes into the "consul member"
* run "yum update"
* update Consul and Nomad binaries, and then reboots the machines
  (the update may have updated the kernel)

Now I am left with services duplicated, one green and one orange for each service.
😭 😭 😭 

Is there a clean way of doing a clean reset of the entire cluster to get rid of the
previous service entries.
Somehow make Nomad and Consul forget all services?

I tried "stop -purge" of all services, but when I submit the jobs, the same "1 green 1 orange" problem surfaces.

I have about 10 application setups running, each with about 15 services.

I plan to:
list out the currently running services (to be started later)
stop all Consuls and Nomad services
remove the data dirs for all Consuls and Nomads
(basically "reformat" the cluster and submit all the jobs again)

Do let me know if there is anything that can be done to recover cleanly.
Thankfully this is only a test setup!

Regards,
Shantanu

**Comment by preetapan** [comment]
@shantanugadgil Had some questions to help us debug this more:

- Are those services that got deduplicated from tasks you run in Nomad, or are you also seeing duplicate services named `nomad` and `nomad-client` in Consul ?
- Can you share your Nomad server and client configs? 
- Do you point more than one nomad client at the same consul agent? The syncing behavior introduced in 0.8 will cause issues if that's the case (details in https://github.com/hashicorp/nomad/issues/4256). That's because Nomad will remove any services from the local agent that are managed on behalf of tasks it doesn't recognize. This can cause repeated delete/add of services registered on behalf of tasks by Nomad if more than one Nomad client is talking to the same Consul agent. 
- Info from one of the duplicated services in Consul would be helpful. Could you share relevant output of v1/agent/services showing the duplicated services?

**Comment by dadgar** [comment]
@shantanugadgil Here is a Linux AMD64 binary to test! [nomad.zip](https://github.com/hashicorp/nomad/files/2074439/nomad.zip)

**Comment by shantanugadgil** [comment]
hi @preetapan @dadgar 
I have rolled out the new binary on my cluster and services are stable for a few hours.
_No more ghost services!_ 👻 

Unfortunately I had started the cleanup (rm data_dir) before I saw the comments.

All my Nomad agent configs have always pointed to consul on `127.0.0.1:8500`, This cluster (albeit growing over time) has been in use since  0.6.x. with the same config.

Would the client and server configs be still required?

Thanks a lot for the quick fix!

Cheers,
Shantanu
